pkg/kubectl/resource/builder.go

```diff
@@ -757,3 +774,17 @@ func HasNames(args []string) (bool, error) {
 	}
 	return hasCombinedTypes || len(args) > 1, nil
 }
+
+func (b *Builder) validateNamespace(c RESTClient, namespace string) error {
+	if !b.validateNamespaces {
+		return nil
+	}
+	if namespace == api.NamespaceDefault || namespace == api.NamespaceAll {
+		return nil
+	}
+	err := c.Get().AbsPath("api", "v1", "namespaces", namespace).Do().Error()
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
refactored away, only handling on 404 for now, all other errors will be dropped.